work_queue/src/work_queue.c

```diff
@@ -5153,6 +5181,13 @@ static work_queue_task_state_t change_task_state( struct work_queue *q, struct w
 			push_task_to_ready_list(q, t);
 			break;
 		case WORK_QUEUE_TASK_DONE:
+			s = malloc(sizeof(struct work_queue_stats));
+			work_queue_get_stats(q, s);
+			//if(t->result == WORK_QUEUE_RESULT_SUCCESS) {
+			//	compute_capacity(q, s);
+			//}
+			log_queue_stats(q);
+			break;
```

Comment:
Move the line log_queue_stats(q) to after the switch.  Remove anything that deals with s (including the definition above), as those lines do not have any effect. Remove the commented code.